src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    }
+  }
```

Comment:
[FUNCTION NAMING] - The `hide` method should only set the `hidden` property to `true`. Currently, it is conditionally toggling the value, which is not in line with the task specification. Consider removing the condition check and directly setting `this.hidden` to `true`.